### DISCUSSION TITLE
Using prisma within a node module, import prisma client fails

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [prisma/prisma#13206] [Q&A] [answered]
Good day..

im working on a project, and using prisma which is great !
im just having abit of trouble getting my node module to work , and was wondering if someone could help me along abit,,
im having trouble with prisma generate and prisma client, 
when i add the module to my testApp prisma client does not seem to find the prisma instance and asks to do a prisma generate..
when i do this before the prisma client is imported, it fails and keeps saying i need to run prisma generate. im abit confuced on how this is. 

im not quite sure what prisma generate does,  vs prisma migrate.. and why it needs it before the import o prisma client, i cant really get this from the docs either as there a bit scattered on this subject.. 

for structure 

app = node_module using prisma
testAPP  uses this module by local yarn add

i would like to keep the migration directory and schema.prisma file inside the node_module  

when i run the test app  it runs prisma generate, but then installs prisma in my testapp  which is not what needs to happen as it needs to stay in its own module.. and then says that it can import prisma client and need to run prisma generate again.. 
how does prisma generate work...? what does it do..? how to get around this problem..?

how would one achieve that everything stays inside the module..?

kind regards....

**Comment by nurul3101** [comment, marked as answer]
Hey @cpt-n3mo 👋 

What version of `@prisma/client` are you using?
When you execute `npx prisma generate` do you get the output that PrismaClient is generated?
Are you able to see the `.prisma/client` folder in your node_modules?

Why would you like to keep schema.prisma and migrations directory in node_modules? These are supposed to be kept in source control and committed.

Could you please elaborate on what you have tried and what output do you get when you run the prisma commands?

**Reply by cpt-n3mo** [reply]
Good day. thank you for your reply. 

i think i just solved the problem 
``` javascript
execSync(`npx prisma generate schema=${join(process.cwd() + '/schema.prisma')}`, { cwd: join(__dirname + "/../"), shell: true, stdio: "inherit" })
```

running this before the import seems to work.. execSync cwd needs to point to the modules root dir, 
and then prisma generate should be run with the location of the schema.prisma where ever that my be. within module or within appDir using the module..

i will mark this as closed solved

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
